### PR TITLE
Fixed changes based upon Xcode analyzer

### PR DESCRIPTION
**Pull request by mikedinicola**
These memory errors are uncovered when running static analysis. Adding nullability annotations should not have side effect as far as I'm aware.

### CODE REVIEW

**Comment by cocojoe** [comment]
Thanks @mikedinicola, apologies for the delay in response. The static analyzer explicitly recommends these changes?

**Comment by mikedinicola** [comment]
No problem, this is not urgent.

As far as the changes, what's important is that there be an annotation. If you prefer to use `nonnull` or `null_unspecified` that would be fine as well. On closer inspection `nonnull` makes more sense in the case of `keyRefOfRSAKeyWithTag`

**Comment by mikedinicola** [comment]
Wow, I haven't had enough caffeine today- the original suggestions definitely made more sense, as that method can clearly return NULL. Reverting, sorry for the mess

File: SimpleKeychain/A0SimpleKeychain+KeyPair.h [hunk 1]
```diff
@@ -74,7 +74,7 @@ NS_ASSUME_NONNULL_BEGIN
  *
  *  @return SecKeyRef of RSA Key
  */
-- (SecKeyRef)keyRefOfRSAKeyWithTag:(NSString *)keyTag;
+- (nullable SecKeyRef)keyRefOfRSAKeyWithTag:(NSString *)keyTag;
```
Comment (cocojoe) [inline SimpleKeychain/A0SimpleKeychain+KeyPair.h]: Please update the doc header to reflect changes.

File: SimpleKeychain/A0SimpleKeychain.h [hunk 2]
```diff
@@ -215,7 +215,7 @@ NS_ASSUME_NONNULL_BEGIN
  *
  *  @return if the value was saved it will return YES. Otherwise it'll return NO.
  */
-- (BOOL)setData:(NSData *)data forKey:(NSString *)key promptMessage:(nullable NSString *)message;
+- (BOOL)setData:(nullable NSData *)data forKey:(nullable NSString *)key promptMessage:(nullable NSString *)message;
```
Comment (cocojoe) [inline SimpleKeychain/A0SimpleKeychain.h]: I don't see this in the analyzer Xcode 10.2

File: SimpleKeychain/A0SimpleKeychain.m [hunk 3]
```diff
@@ -115,7 +115,7 @@ - (BOOL)setString:(NSString *)string forKey:(NSString *)key promptMessage:(NSStr
 }
 
 
-- (BOOL)setData:(NSData *)data forKey:(NSString *)key promptMessage:(NSString *)message {
+- (BOOL)setData:(nullable NSData *)data forKey:(nullable NSString *)key promptMessage:(NSString *)message {
```
Comment (cocojoe) [inline SimpleKeychain/A0SimpleKeychain.m]: I don't see this in the analyzer Xcode 10.2

**Comment by cocojoe** [comment]
Outdated

**Pull request closed without merging**